api/nim/v1/account_types.go

```diff
@@ -11,6 +11,8 @@ import (
 type AccountSpec struct {
 	// A reference to the Secret containing the NGC API Key.
 	APIKeySecret corev1.ObjectReference `json:"apiKeySecret"`
+	// A reference to the ConfigMap containing the list of NIM models that are allowed to be deployed.
+	ModelListConfig *corev1.ObjectReference `json:"modelListConfig,omitempty"`
```

Comment:
Maybe we should set a default name instead if _omitempty_, so admins will have to create the ConfigMap but can skip updating the Account?

Reply:
I don't think so.
There will be an error, when it is set in the spec, but the config map doesn't exist. Although, the controller will not requeue in this case.
I don't think we should leave an `Account` in an erroneous situation by setting a default value in the spec. 

You suggested to not requeue when there is anything wrong with the config map, in order to avoid infinite loop. And I did that. 
But, IMHO, it should be requeued if there is anything wrong with the spec. That's how the controller is supposed to work.
The dashboard should deal with the infinite loop, not the backend. But I don't want to insist on it.

Reply:
IMO if we change this [log record](https://github.com/xieshenzh/odh-model-controller/blob/nim-model-list/internal/controller/nim/account_controller.go#L464) from _Error_ to _Info_, we should be good with setting default file name, and if we don't already cache this ConfigMap, we can do that too.

Anyway, it was just a thought. Feel free to resolve this if you disagree. :-)